### DISCUSSION TITLE
Keyed Services Support in MvvmToolkit.Mvvm

### DISCUSSION THREAD

**Discussion by Touseefelahi** [CommunityToolkit/dotnet#800] [Q&A]
How to use keyed services in my Avalonia Application?
I am trying to use the keyed services feature from .net 8, but it is not working for me.

This is what I have in my App.axaml.cs file
```
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using CommunityToolkit.Mvvm.DependencyInjection;

var services = new ServiceCollection();
services.AddKeyedTransient<IUnitOfWork, UnitOfWork>(ServiceList.LocalDB);
Ioc.Default.ConfigureServices(ConfigureServices());
```
             
             
This is how I am trying to get the keyed service in some other class
```
Ioc.Default.GetRequiredKeyedService<IUnitOfWork>("LocalDB");
```
This is the error that I am getting:

`This service provider doesn't support keyed services.
`            

Am I missing something here? or the feature is not implemented yet?

Thank you

**Comment by aodpi** [comment]
I think it's not implemented yet and would be nice to have, but in the meantime you can do it like this:
```csharp
var sp = IoC.Default.GetRequiredService<IServiceProvider>();
var service = sp.GetRequiredKeyedService<IUnitOfWork>("LocalDB");
```
